pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -170,7 +170,7 @@ func (d *DependenciesDistributor) Reconcile(key util.QueueKey) error {
 	}
 	klog.V(4).Infof("DependenciesDistributor start to reconcile object: %s", clusterWideKey)
 
-	bindingObjectList, err := d.resourceBindingLister.ByNamespace(clusterWideKey.Namespace).List(labels.Everything())
+	bindingObjectList, err := d.resourceBindingLister.List(labels.Everything())
```

Comment:
Hi @yanfeng1992, I think you have found the root cause of the problem. 

But this change can have a big impact, because the total number of `ResourceBinding` may be large, which makes me a little hesitant to support this feature (Yes, I think the current patch is a new feature, not a bug fix), or we need to add a flag to enable this feature.

Ask more people for more ideas. @RainbowMango @mrlihanbo @jwcesign

Reply:
I more like giving some labels to the dependency resource's rb, then using label selector to select the bindingObject list.

Reply:
Hi @yanfeng1992, how about finding the target `ResourceBinding` through the following path?

resource template -> ResourceBinding of resource template -> dependent ResourceBinding

Reply:
> Hi @yanfeng1992, how about finding the target `ResourceBinding` through the following path?
> 
> resource template -> ResourceBinding of resource template -> dependent ResourceBinding

i try to find bindobject.Find dependent ResourceBinding according to bindingObject.Spec.RequiredBy, is there anything wrong? @XiShanYongYe-Chang

Reply:
The newly added e2e seems to run failed.

Reply:
You can get the log from here: https://github.com/karmada-io/karmada/suites/10409837217/artifacts/514337324

At the bottom of this page: https://github.com/karmada-io/karmada/actions/runs/3937200810